BooksScreen.tsx

```diff
@@ -1,37 +1,46 @@
-import React, { useEffect } from 'react';
-import { Alert, FlatList, StyleSheet, Text, View } from 'react-native';
-
-import { setBooks } from '@core/services/bookSlice';
-import { useSelector } from 'react-redux'
+import React, { useEffect } from "react";
+import {
+  FlatList,
+  StyleSheet,
+  Text,
+  View,
+  TouchableOpacity,
+} from "react-native";
+import { useNavigation } from "@react-navigation/native";
+import { useSelector, useDispatch } from "react-redux";
+import { getBooks } from "./booksApi";
+import { updateFavourite } from "./bookSlice";
 
 interface BookItemProps {
   book: Book;
 }
 
 const BooksScreen = (): JSX.Element => {
   const books = useSelector((state) => state.books);
-
+  const dispatch = useDispatch();
+  const navigation = useNavigation();
   useEffect(() => {
     getBooks();
   }, []);
 
-  async function getBooks() {
-    try {
-      const response = await fetch('https://example.com/api/books');
-      const books: Book[] = await response.json();
-      setBooks(books);
-    } catch (e) {
-      Alert.alert('Fetch books failed with error: ', e);
-    }
-  }
+  const changeFavourite = (id, favourite) => {
+    dispatch(updateFavourite({ id, favourite }));
+  };
 
   const BookItem = ({ book }: BookItemProps) => {
     return (
       <View>
-        <View>
+        <TouchableOpacity
+          onPress={() => navigation.navigate("BooksDetail", { id: book.id })}
+        >
```

Comment:
Instead of passing as params, lets dispatch redux action `selectBook(id)`, store `selectedBookId` in redux store